### ISSUE TITLE
App startup fails with black screen

### ISSUE THREAD

**Issue by hsjoberg** [hsjoberg/blixt-wallet#155]
Seems to happen on Samsung devices. Cause is still unknown.

~It could be related to lnd failing to start up if the app/phone does not have an internet connection as Neutrino will not be able to proceed with the current config (possible fix is using `neutrino.addpeer` instead of `neutrino.connect` as the latter mandates that the peer is successfully connected to).~

Needs to be fixed before release.

**Comment by hsjoberg** [comment]
In general better lnd failure detection is needed, opening another issue about that #156, although that will take some work to implement.

**Comment by hsjoberg** [comment]
It could also be related to scheduled job being stale and messing up with MainActivity startup, and thus not related or semi-related to lnd.

**Comment by hsjoberg** [comment]
I've yet to fully confirm this, but It seems to be related to scheduled sync jobs. `syncedToGraph` never reaches true, which keeps the work stale for seemingly forever.  
For some reason this keeps the app from starting properly and instead goes to a black screen (probably stays on `Init.tsx`). (Perhaps `start` or `checkStatus` in LndMobile fails.)

An immediate fix is just to wait for `syncedToSync` to be true and skip graph sync as it looks like lnd have faulty behavior.

**Comment by hsjoberg** [comment]
This indicates that it's a race condition happening in the AsyncStorage SQlite backend as both scheduled sync and main app uses it:

```
04-24 10:08:42.022 17527 17527 I LndMobile: Service attached
04-24 10:08:42.023 17527 20728 E SQLiteLog: (10) POSIX Error : 11 SQLite Error : 3850
04-24 10:08:42.023 17527 20728 E SQLiteLog: (10) LOCK : Getting pending lock for (1) failed due to (11). The current owner is (20781).
04-24 10:08:42.023 20781 20781 D LndMobileService: New incoming message from client, msg id: 1
04-24 10:08:42.023 20781 20781 D LndMobileService: Got register client android.os.Messenger@a0b6efe
04-24 10:08:42.023 17527 20715 D LndMobile: New incoming message from LndMobileService, msg id: 2
04-24 10:08:42.023 17527 20715 E LndMobile: Unknown request: -1462989363
04-24 10:08:42.023 17527 20715 E AndroidRuntime: FATAL EXCEPTION: mqt_native_modules
04-24 10:08:42.023 17527 20715 E AndroidRuntime: Process: com.blixtwallet, PID: 17527
04-24 10:08:42.023 17527 20715 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke interface method 'void com.facebook.react.bridge.Promise.resolve(java.lang.Object)' on a null object reference
04-24 10:08:42.023 17527 20715 E AndroidRuntime: 	at com.blixtwallet.LndMobile$a.handleMessage(Unknown Source:369)
04-24 10:08:42.023 17527 20715 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:107)
04-24 10:08:42.023 17527 20715 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:237)
04-24 10:08:42.023 17527 20715 E AndroidRuntime: 	at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(Unknown Source:37)
04-24 10:08:42.023 17527 20715 E AndroidRuntime: 	at java.lang.Thread.run(Thread.java:919)
04-24 10:08:42.024 17527 20728 E SQLiteLog: (10) POSIX Error : 11 SQLite Error : 3850
04-24 10:08:42.024 17527 20728 E SQLiteLog: (10) LOCK : Getting pending lock for (1) failed due to (11). The current owner is (20781).
04-24 10:08:42.026  6576 20820 I DropBoxManagerService: add tag=data_app_crash isTagEnabled=true flags=0x2
04-24 10:08:42.029  6576  6649 W BroadcastQueue: Background execution not allowed: receiving Intent { act=android.intent.action.DROPBOX_ENTRY_ADDED flg=0x10 (has extras) } to com.google.android.gms/.stats.service.DropBoxEntryAddedReceiver
04-24 10:08:42.029  6576  7392 D Debug   : !@DumpState : SHIP
04-24 10:08:42.030  6576  7392 D Debug   : !@DumpState : debug level:0x4f4c
04-24 10:08:42.030  6576  7392 D Debug   : !@Dumpstate : Finally, system will skip dumpstate
04-24 10:08:42.030  6576  7392 W ActivityManager: crash : com.blixtwallet,0
04-24 10:08:42.031  6576  7392 W ActivityTaskManager:   Force finishing activity com.blixtwallet/.MainActivity
```

**Comment by hsjoberg** [comment]
Hopefully fixed in 0194418c531a620b7398b6568779340d437f2d1d and b9076441edaf422227cf9415f372ebad251d9d28

What seems to happen is that both the app and the sync job tries to either start up lnd or connect to the IPC bridge at the same time, which causes problems.  
As a mitigation the sync job will not check if the app is started and immediately close down if it is.

**Comment by hsjoberg** [comment]
This issue can possibly be related to https://github.com/oblador/react-native-keychain/issues/337
Related Blixt issue https://github.com/hsjoberg/blixt-wallet/issues/184

One user are seeing freeze for >5 minutes, it's plausible this has can happen on S20 Ultra as well.
I never let the app keep going when I conducted tests because I assumed something wrong was going on with LndMobile.
